src/preset_cli/cli/superset/sync/dbt/metrics.py

```diff
@@ -59,3 +70,47 @@ def apply_filters(sql: str, filters: List[FilterSchema]) -> str:
         "{field} {operator} {value}".format(**filter_) for filter_ in filters
     )
     return f"CASE WHEN {condition} THEN {sql} END"
+
+
+def is_derived(metric: MetricSchema) -> bool:
+    """
+    Return if the metric is derived.
+    """
+    return (
+        metric.get("calculation_method") == "derived"  # dbt >= 1.3
+        or metric.get("type") == "expression"  # dbt < 1.3
+    )
+
+
+def get_metrics_for_model(
+    model: ModelSchema,
+    metrics: List[MetricSchema],
+) -> List[MetricSchema]:
+    """
+    Given a list of metrics, return those that are based on a given model.
+    """
+    metric_map = {metric["unique_id"]: metric for metric in metrics}
+    related_metrics = []
+
+    for metric in metrics:
+        parents = set()
+        queue = [metric]
+        while queue:
+            node = queue.pop()
+            depends_on = node["depends_on"]
+            if is_derived(node):
+                queue.extend(metric_map[parent] for parent in depends_on)
```

Comment:
This is the crux of the PR: for derived metrics we need to look at the upstream metrics to find the parent models.

Reply:
is the metric list something from superset or something from the dbt manifest?